### PR TITLE
Add hazard generation

### PR DESCRIPTION
**Pull request by roguefirework**
See title

### CODE REVIEW

**Review by MaximumMaxxx — CHANGES_REQUESTED** [review]
Typo in docs, please fix

**Review by MaximumMaxxx — APPROVED** [review]
I agree

File: Core/WumpusCore/GameLocations/GameLocations.cs [hunk 1]
```diff
@@ -36,14 +37,50 @@ public RoomType[] Rooms
...
-        public GameLocations(ushort numRooms)
+        /// <param name="numVats">The number of vat rooms to generate</param>
+        /// <param name="numBats">The number of bat rooms to generate</param>
+        /// <param name="numRats">The number of rat rooms to generate</param>
+        /// <param name="numAcrobats">The number of acrobat rooms to generate</param>
+        /// <param name="topology">The topology structure</param>
+        /// <param name="random">A random object</param>
+        public GameLocations(ushort numRooms,ushort numVats, ushort numBats, ushort numRats, ushort numAcrobats, Topology.Topology topology, Random random)
         {
+            if (numVats + numRats + numAcrobats + numBats >= 30)
```
Comment (1118422ILHS) [inline Core/WumpusCore/GameLocations/GameLocations.cs]: Replace 30 with topology.numRooms

**Review by 1118422ILHS — CHANGES_REQUESTED** [review]
Test it.

**Review by 1118422ILHS — CHANGES_REQUESTED** [review]
No more ushorts

File: Core/WumpusTesting/ControllerTest.cs [hunk 2]
```diff
@@ -9,6 +9,7 @@ public class ControllerTest
         [TestMethod]
         public void testMap()
```
Comment (1118422ILHS) [inline Core/WumpusTesting/ControllerTest.cs:10]: whurg?

File: Core/WumpusCore/Topology/Graph.cs
Comment (ilhs1096069) [inline Core/WumpusCore/Topology/Graph.cs:1]: Please add some docs to this file :(

**Review by ilhs1096069 — CHANGES_REQUESTED** [review]
Please add docs to the file I said to

**Review by 1118422ILHS — APPROVED** [review]
Appears exquisite to the eyes.

**Review by ilhs1096069 — APPROVED** [review]
Yippee it looks good now thank you

**Pull request merged**